compiler/backend/jsflow.nim

```diff
@@ -0,0 +1,361 @@
+## Implements the translation of CGIR control-flow constructs to JavaScript
+## constructs.
+##
+## While JavaScript doesn't directly support the kind of control flow that
+## the MIR supports, it does support 'finally' and 'catch', both which are
+## leveraged for the translation. The idea: figure out which statements
+## JavaScript blocks, 'finally's, and 'catch's need to enclose and then place
+## them in a way such that the behaviour is translated correctly.
+
+import
+  std/[
+    packedsets,
+    tables
+  ],
+  compiler/backend/[
+    cgir
+  ],
+  compiler/utils/[
+    idioms
+  ]
+
+type
+  StructKind* = enum
+    stkTry   ## start of a 'try' statement
+    stkBlock ## start of a labeled block
+
+    stkStructStart ## start of an 'if' or 'while'
+    stkCatch   ## a 'catch' clause of a 'try' statement
+    stkFinally ## a 'finally' clause of a 'try' statement
+
+    stkEnd     ## end of a catch, finally, or block
```

Comment:
AFAICT this also marks the end of a `stkStructStart`?

Reply:
Good catch, it does. I'll update the comment.